internal/runners/branch/tree.go

```diff
@@ -0,0 +1,192 @@
+package branch
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+	"github.com/go-openapi/strfmt"
+)
+
+type branchNode struct {
+	Label string
+	ID    string
+}
+
+type tree map[branchNode]tree
```

Comment:
@daved is this "proper" Go? Feels really weird how it's self referencing.

Reply:
Yep! It's not something I've needed to do often, so I can't offer much in the way of guidance regarding infinite recursion except that (IIRC) I've tended toward checking that on insertion. Though, I suppose that may be swayed by read versus write priority.